torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -2827,7 +2857,7 @@ void SegmentCandidateFinder::findSegments() {
 
         auto candidate_it = candidates.begin();
         while (candidate_it != candidates.end() &&
-               !codeGenSupportedMerge(candidate_it->edge)) {
+               !codeGenSupportedMerge(group, candidate_it->group)) {
```

Comment:
This change is counter-intuitive.

The old API only tries to merge all candidate connected by `candidate_it->edge`. Shouldn't we plumb the connection check to `getMergeCandidates` instead of assert on the connectivity inside `codeGenSupportedMerge`?